Description: Keep old MediaKeys D-Bus interface since a lot of stuff
 still uses it. Consider dropping this patch after 18.04 LTS.
 https://mail.gnome.org/archives/distributor-list/2017-April/msg00009.html
Author: Jeremy Bicha <jbicha@ubuntu.com>
---
 plugins/media-keys/gsd-media-keys-manager.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/plugins/media-keys/gsd-media-keys-manager.c
+++ b/plugins/media-keys/gsd-media-keys-manager.c
@@ -223,6 +223,7 @@ struct GsdMediaKeysManagerPrivate
 
         /* Multimedia keys */
         guint            mmkeys_name_id;
+        guint            gsd_name_id;
         MprisController *mpris_controller;
 };
 
@@ -2929,6 +2930,11 @@ gsd_media_keys_manager_stop (GsdMediaKey
                 priv->mmkeys_name_id = 0;
         }
 
+        if (priv->gsd_name_id > 0) {
+                g_bus_unown_name (priv->gsd_name_id);
+                priv->gsd_name_id = 0;
+        }
+
         if (priv->bus_cancellable != NULL) {
                 g_cancellable_cancel (priv->bus_cancellable);
                 g_object_unref (priv->bus_cancellable);
@@ -3243,6 +3249,11 @@ on_bus_gotten (GObject             *sour
                                                                       G_BUS_NAME_OWNER_FLAGS_NONE,
                                                                       NULL, NULL, NULL, NULL);
 
+        manager->priv->gsd_name_id = g_bus_own_name_on_connection (manager->priv->connection,
+                                                                   "org.gnome.SettingsDaemon",
+                                                                   G_BUS_NAME_OWNER_FLAGS_NONE,
+                                                                   NULL, NULL, NULL, NULL);
+
         g_dbus_proxy_new (manager->priv->connection,
                           G_DBUS_PROXY_FLAGS_NONE,
                           NULL,
